src/context.rs

```diff
@@ -565,11 +560,18 @@ impl DefaultSeafowlContext {
     // Check that the TableReference doesn't have a database/schema in it.
     // We create all external tables in the staging schema (backed by DataFusion's
     // in-memory schema provider) instead.
-    fn resolve_staging_ref<'a>(
-        &'a self,
-        name: &'a str,
-    ) -> Result<ResolvedTableReference<'a>> {
-        let reference: TableReference = name.into();
+    fn resolve_staging_ref(
+        &self,
+        name: &OwnedTableReference,
+    ) -> Result<OwnedTableReference> {
+        // NB: Since Datafusion 16.0.0 there's this OwnedTableReference enum and for external tables
+        // the parsed ObjectName (which may be multipart, fully-qualified name) is coerced into the
+        // `Bare` enum variant (see `external_table_to_plan` in datafusion-sql) for some reason.
+        //
+        // This means that any potential catalog/schema references get condensed into the name, so
+        // we have to unravel that name here again, and then resolve it properly.
+        let full_name = name.to_string();
```

Comment:
Due to a datafusion bug (still unfixed in 16.0), calling `ObjectName.to_string()` will double quote names with special chars. Instead, we must do:
```rust
pub fn remove_quotes_from_ident(possibly_quoted_name: &Ident) -> Ident {
    Ident::new(&possibly_quoted_name.value)
}

pub fn remove_quotes_from_idents(column_names: &[Ident]) -> Vec<Ident> {
    column_names.iter().map(remove_quotes_from_ident).collect()
}

pub fn remove_quotes_from_object_name(name: &ObjectName) -> ObjectName {
    ObjectName(remove_quotes_from_idents(&name.0))
}

// ...
let full_name = remove_quotes_from_object_name(&name).to_string()
```

These fuctions are added to `context.rs` in [PR 269](https://github.com/splitgraph/seafowl/pull/269).

Reply:
Understood, though note that the `name` above is actually not an `ObjectName`, but an `OwnedTableReference` instead, and it's [`to_string()`](https://github.com/apache/arrow-datafusion/blob/fad77a4b278a441e343fb6caa76f6e3bcd27e248/datafusion/common/src/table_reference.rs#L107-L121) shouldn't require unquoting right?

Regardless, I'll wait for your PR to merge before merging this one.